cadc-pkg-server/src/main/java/org/opencadc/pkg/server/PackageRunner.java

```diff
@@ -0,0 +1,260 @@
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 5 $
+*
+************************************************************************
+*/
+
+package org.opencadc.pkg.server;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.io.ByteCountOutputStream;
+import ca.nrc.cadc.log.WebServiceLogInfo;
+import ca.nrc.cadc.rest.SyncOutput;
+import ca.nrc.cadc.util.ThrowableUtil;
+import ca.nrc.cadc.uws.ErrorSummary;
+import ca.nrc.cadc.uws.ErrorType;
+import ca.nrc.cadc.uws.ExecutionPhase;
+import ca.nrc.cadc.uws.Job;
+import ca.nrc.cadc.uws.ParameterUtil;
+import ca.nrc.cadc.uws.server.JobRunner;
+import ca.nrc.cadc.uws.server.JobUpdater;
+import ca.nrc.cadc.uws.util.JobLogInfo;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.URI;
+import java.net.URL;
+import java.util.Date;
+import java.util.Iterator;
+import org.apache.log4j.Logger;
+
+public abstract class PackageRunner implements JobRunner
+{
+    private static final Logger log = Logger.getLogger(PackageRunner.class);
+
+    protected Job job;
+
+    protected JobUpdater jobUpdater;
+    protected SyncOutput syncOutput;
+    protected ByteCountOutputStream counter;
+    protected WebServiceLogInfo logInfo;
+
+    public PackageRunner() 
+    {
+    }
+
+    /**
+     * Generate the package name
+     * @return String with package name.
+     */
+    protected abstract String generatePackageName();
```

Comment:
`getPackageName()`??

Also (relateed to this method): you are going to want code like that in `initTarResponse()` to set headers, inc. calling getPackageName() for the `content-disposition` header. I mention it here because PackageRunner being responsible for setting headers before opening stream is the reason this abstract method exists.

Reply:
- 'generatePackageName' is from the original code, as the filename is 'generated' from using a static string + the job id. (cadc-download-<jobid>.tar) - Doesn't matter if it's get instead...

- content-disposition header content will depend on what kind of package is being written.  
tar: Content-Disposition = "inline;filename=*.tar"
zip: Content-Disposition="attachment;filename=*.zip"

... if I've read about this correctly. At the least the file extension will differ. 

So either the whole function is provided by extending classes, or there's variables that need to be populated. Which way to go?

Reply:
In last week's slack thread we said `getPackageName()`. 
I would say subclass (impl) should provide the base filename (sans extension) and it can decide to include something to avoid filename collisions (jobid, timestamp, or something  meaningful like caom2-packlage-server did when there was a single input ID). 
PackageRunner should append the correct extension consistent with the `content-type`.

Reply:
The Apache Commons Compress library automatically generates an extension for any files it generates. 

Content disposition filename extension needs to match that. So it's a value that should come from the Writer class (TarWriter or ZipWriter.)

Reply:
changed to getPackageName(), added a function to set up the OutputStream with the correct disposition string, content-type is hard coded in PackageRunner for now as only tar is currently supported. It's expected that will change when a mechanism to support requesting tar or zip is implemented.